treeherder/perfalert/perfalert/__init__.py

```diff
@@ -97,16 +97,25 @@ def add_data(self, push_timestamp, value, **kwargs):
         self.data.append(Datum(push_timestamp, value, **kwargs))
         self.data.sort()
 
-    def analyze_t(self, back_window=12, fore_window=12, t_threshold=7):
+    def analyze_t(self, min_back_window=12, max_back_window=24,
+                  fore_window=12, t_threshold=7):
         # Use T-Tests
         # Analyze test data using T-Tests, comparing data[i-j:i] to data[i:i+k]
-        (j, k) = (back_window, fore_window)
+        (j, k) = (min_back_window, fore_window)
         good_data = []
 
         num_points = len(self.data) - k + 1
+        last_seen_regression = 0
         for i in range(num_points):
             di = self.data[i]
-            jw = [d.value for d in good_data[-j:]]
+            # if we haven't seen something that looks like a change in a
+            # while, incorporate some extra historical data into our t-test
+            # calculation
+            if last_seen_regression > (max_back_window - min_back_window):
+                additional_back_window = (max_back_window - min_back_window)
```

Comment:
@klahnakoski pointed out to me that this is probably wrong -- we should gradually reintroduce the larger back window after the regression, something like:

```py
if last_seen_regression > min_back_window:
  additional_back_window = min(last_seen_regression, max_back_window - min_back_window)
```